### PR TITLE
NO-JIRA: extend clusteroperators gatherer to collect status of insightsoperato…

### PR DESCRIPTION
**Pull request by tremes**
…r resource

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

- `docs/insights-archive-sample/config/clusteroperator/operator.openshift.io/insightsoperator/cluster.json`

## Documentation

- `path/to/documentation.md`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: pkg/gatherers/clusterconfig/gather_cluster_operators.go [hunk 1]
```diff
@@ -185,14 +188,25 @@ func collectClusterOperatorRelatedObjects(ctx context.Context,
 		if !ok {
 			klog.Warningf("Can't find spec for cluster operator resource %s", clusterResource.GetName())
 		}
+
+		// this is an exception for the insightsoperators.operator.openshift.io
+		// "cluster" resource to also collect the status
+		var status interface{}
```
Comment (rluders) [inline pkg/gatherers/clusterconfig/gather_cluster_operators.go:194]: I would probably change it to:

```go
anonymizeIdentityProviders(clusterResource.Object)
coResource := clusterOperatorResource{
	Spec:       spec,
	Kind:       clusterResource.GetKind(),
	Name:       clusterResource.GetName(),
	APIVersion: clusterResource.GetAPIVersion(),
	namespace:  clusterResource.GetNamespace(),
}

if clusterResource.GetKind() == "InsightsOperator" {
	coResource.Status = clusterResource.Object["status"]
}

res = append(res, coResource)
```

**Review by rluders — CHANGES_REQUESTED** [review]

**Review by rluders — APPROVED** [review]
/lgtm

**Comment by JoaoFula** [comment]
https://gitlab.cee.redhat.com/ccx/insights-operator-tests/-/merge_requests/537

**Pull request merged**